dxr/query.py

```diff
@@ -386,12 +400,16 @@ def filter_menu_items(plugins):
     with the languages it supports.
 
     """
+    # Concretize to iterate over it more than once.
+    plugins = list(plugins)
     sorted_filters_by_name = sorted(
-        ((name, filters[0]) for name, filters in filters_by_name(plugins).items()),
-        key=lambda (name, filter): (hasattr(filter, 'lang'), name))
-    return (dict(name=name, description=filter.description)
-            for name, filter in sorted_filters_by_name
-            if filter.description)
+        ((name, filters) for name, filters in filters_by_name(plugins).items()),
+        key=lambda (name, filters): (hasattr(filters[0], 'lang'), name))
+    badge_colors = lang_badge_colors(plugins)
+    return (dict(name=name, description=filters[0].description,
+                 badges=[(f.lang, badge_colors.get(f.lang)) for f in filters if hasattr(f, 'lang')])
```

Comment:
Is there a reason for .get(f.lang) over just [f.lang] since we're already testing hasattr(f, 'lang')?